tools/cordova/package-id-version-parser.js

```diff
@@ -0,0 +1,27 @@
+// Accepts a combined Cordova package ID + version string, then parses out
+// and returns the ID and version in a package details object.
+//
+// Example `packageIdAndVersion` formats:
+// some-cordova-plugin@1.0.0
+// @somescope/some-cordova-plugin@1.0.0
+exports.parse = packageIdAndVersion => {
+  const package = {};
+  if (packageIdAndVersion) {
+    const parts = packageIdAndVersion.match(
+      /^(@[^\/]*)?(\/)?([^@]*)(@)?(.*)?/
+    );
+    const details = {
+      scope: parts[1],
+      scopeSeparator: parts[2],
+      packageName: parts[3],
+      versionSeparator: parts[4],
+      version: parts[5],
+    };
```

Comment:
I could probably go further. Is there a way we could avoid the regex and just use a combination of `packageIdAndVersion.startsWith('cordova:)` and split on `packageIdAndVersion.lastIndexOf('@')`.

Reply:
Given the tests, and that you may have already spent some time on this, and that overall this is an improvement, I think it's okay as is too if you just wanted to put your hands up. 🙌 

:wink:

Reply:
I originally avoided the regex and did just that, but the problem is that we need to catch instances where the version number isn't passed in. So the `parse` function can actually have `@somescope/some-cordova-plugin` passed in, which means splitting on `@` doesn't quite work (since we can have both a valid and invalid package string passed in that has only one `@`). We can definitely add code to address this, but it's at this point I decided a regex would probably make more sense. Happy to change it if you disagree though!

Reply:
Makes sense to me.  Thanks for the explanation!